.github/actions/upload/src/inputs.js

```diff
@@ -0,0 +1,161 @@
+import { resolve } from 'node:path'
+import { ethers } from 'ethers'
+import { ERROR_CODES, FilecoinPinError, getErrorMessage } from './errors.js'
+
+/**
+ * @typedef {import('./types.js').ParsedInputs} ParsedInputs
+ */
+
+/**
+ * Check if object has own property
+ * @param {any} object - Object to check
+ * @param {string} key - Property key
+ * @returns {boolean} True if object has own property
+ */
+const own = (object, key) => Object.hasOwn(object, key)
+
+/** @type {any} */
+let cachedInputsJson
+
+function readInputsJson() {
+  if (cachedInputsJson !== undefined) return cachedInputsJson
+
+  const raw = process.env.INPUTS_JSON
+  if (!raw) {
+    cachedInputsJson = null
+    return cachedInputsJson
+  }
+
+  try {
+    cachedInputsJson = JSON.parse(raw)
+  } catch (error) {
+    throw new Error(`Failed to parse INPUTS_JSON: ${error instanceof Error ? error.message : String(error)}`)
+  }
+
+  return cachedInputsJson
+}
+
+/**
+ * Convert value to string with fallback
+ * @param {any} value - Value to convert
+ * @param {string} fallback - Fallback value
+ * @returns {string} String representation
+ */
+function toStringValue(value, fallback = '') {
+  if (value === undefined || value === null) return String(fallback ?? '')
+  return typeof value === 'string' ? value : String(value)
+}
+
+/**
+ * Get input value from environment variables
+ * @param {string} name - Input name
+ * @param {string} fallback - Default value
+ * @returns {string} Input value
+ */
+export function getInput(name, fallback = '') {
+  const json = readInputsJson()
+  if (json && own(json, name)) {
+    return toStringValue(json[name], fallback).trim()
+  }
+
+  const envKey = `INPUT_${name.toUpperCase()}`
+  if (process.env[envKey] !== undefined && process.env[envKey] !== null) {
+    return toStringValue(process.env[envKey], fallback).trim()
+  }
+
+  return toStringValue(fallback).trim()
+}
+
+/**
+ * Parse boolean value from string
+ * @param {any} v - Value to parse
+ * @returns {boolean} Parsed boolean
+ */
+export function parseBoolean(v) {
+  if (typeof v === 'boolean') return v
+  if (typeof v !== 'string') return false
+  const s = v.trim().toLowerCase()
+  return s === 'true' || s === '1' || s === 'yes'
+}
+
+/**
+ * Parse and validate all action inputs
+ * @param {string} phase - Action phase (compute, from-cache, or upload/single)
+ * @returns {ParsedInputs} Parsed and validated inputs
+ */
+export function parseInputs(phase = 'single') {
+  const walletPrivateKey = getInput('walletPrivateKey')
+  const contentPath = getInput('path')
+  const networkRaw = getInput('network')
+  const minStorageDaysRaw = getInput('minStorageDays', '')
+  const filecoinPayBalanceLimitRaw = getInput('filecoinPayBalanceLimit', '')
+  const withCDN = parseBoolean(getInput('withCDN', 'false'))
+  const providerAddress = getInput('providerAddress', '0xa3971A7234a3379A1813d9867B531e7EeB20ae07')
```

Comment:
I think I saw something about using a fixed provider for this stuff, I know we have more reliable providers at the moment but this just shunts them all to a single provider (plus also the provider's addresses are hardcoded in output.js which doesn't seem ideal since we can get that info out of synapse).

Can we just leave it defaulting to no provider so it'll pick one for the user and stick with that?

We have some redundancy coming soon, so we'll have multiple deals being made in one go but that'll also be a kind of auto-selection as well.